csharp/src/Apache.Arrow.Flight.Sql/Client/FlightSqlClient.cs

```diff
@@ -0,0 +1,1211 @@
+
+        if (transaction == null)
+        {
+            throw new ArgumentNullException(nameof(transaction));
+        }
+
+        try
+        {
+            var actionCommit = new FlightAction("Commit", transaction.TransactionId);
+            return _client.DoAction(actionCommit, options.Headers);
+        }
+        catch (RpcException ex)
+        {
+            throw new InvalidOperationException("Failed to commit transaction", ex);
+        }
+    }
+
+    /// <summary>
+    /// Rollback a transaction.
+    /// After this, the transaction and all associated savepoints will be invalidated.
+    /// </summary>
+    /// <param name="options">RPC-layer hints for this call.</param>
+    /// <param name="transaction">The transaction to rollback.</param>
+    /// <returns>A Task representing the asynchronous operation.</returns>
+    public AsyncServerStreamingCall<FlightResult> RollbackAsync(FlightCallOptions options, Transaction transaction)
+    {
+        if (options == null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        if (transaction == null)
+        {
+            throw new ArgumentNullException(nameof(transaction));
+        }
+
+        try
+        {
+            var actionRollback = new FlightAction("Rollback", transaction.TransactionId);
+            return _client.DoAction(actionRollback, options.Headers);
+        }
+        catch (RpcException ex)
+        {
+            throw new InvalidOperationException("Failed to rollback transaction", ex);
+        }
+    }
+
+    /// <summary>
+    /// Create a prepared statement object.
+    /// </summary>
+    /// <param name="options">RPC-layer hints for this call.</param>
+    /// <param name="query">The query that will be executed.</param>
+    /// <param name="transaction">A transaction to associate this query with.</param>
+    /// <returns>The created prepared statement.</returns>
+    public async Task<PreparedStatement> PrepareAsync(FlightCallOptions options, string query,
+        Transaction? transaction = null)
+    {
+        transaction ??= NoTransaction();
+
+        if (options == null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        if (string.IsNullOrEmpty(query))
+        {
+            throw new ArgumentException("Query cannot be null or empty", nameof(query));
+        }
+
+        try
+        {
+            var preparedStatementRequest = new ActionCreatePreparedStatementRequest
+            {
+                Query = query,
+                TransactionId = transaction.IsValid()
+                    ? ByteString.CopyFromUtf8(transaction.TransactionId)
+                    : ByteString.Empty,
+            };
+
+            var action = new FlightAction(SqlAction.CreateRequest, preparedStatementRequest.PackAndSerialize());
+            var call = _client.DoAction(action, options.Headers);
+
+            await foreach (var result in call.ResponseStream.ReadAllAsync())
```

Comment:
Add `ConfigureAwait(false)` if supported (I've not tried on a foreach)

Reply:
done